dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryAPITest.java

```diff
@@ -664,4 +668,66 @@ protected static Structure createStructure ( String name, String structureVeloci
         return testStructure;
     }
 
+    @Test
+    public void testDuplicatedCategories() {
+
+        final CategoryAPI categoryAPI = APILocator.getCategoryAPI();
+
+        Category category = null;
+
+        try {
+            //Test varName with proper camel case.
+            final String categoryVarName = "categoryVarNameToTest";
+            String suggestedCategoryVarName = categoryAPI
+                    .suggestVelocityVarName(categoryVarName);
+
+            assertEquals(categoryVarName, suggestedCategoryVarName);
+
+            //Test varName with spaces and 1st letter uppercase.
+            final String categoryVarNameWithSpaces = "Category Var Name To Test";
+            suggestedCategoryVarName = categoryAPI
+                    .suggestVelocityVarName(categoryVarNameWithSpaces);
+
+            assertEquals(categoryVarName, suggestedCategoryVarName);
+
+            //Test varName with spaces and no uppercase.
+            final String categoryVarNameWithSpacesNouppercase = "category var name to test";
+            suggestedCategoryVarName = categoryAPI
+                    .suggestVelocityVarName(categoryVarNameWithSpacesNouppercase);
+
+            assertEquals(categoryVarName, suggestedCategoryVarName);
+
+            //Now lets create a Category to check how we handle duplicated varNames.
+            category = new Category();
+            category.setCategoryName("Category Var Name To Test");
+            category.setKey("categoryNameWithSpaces-1");
+            category.setCategoryVelocityVarName(categoryVarName);
+            category.setSortOrder((String) null);
+            category.setKeywords(null);
+
+            categoryAPI.save(null, category, user, false);
+
+            Category foundCategory = categoryAPI.find(category.getCategoryId(), user, false);
+            assertNotNull(foundCategory);
```

Comment:
As a suggestion, I would personally remove these asserts, since we assume the categoryAPI.find method works and is outside the scope of this test